python/lsst/rubintv/production/rubinTv.py

```diff
@@ -554,3 +631,161 @@ def run(self):
         """Run continuously, calling the callback method on the latest dataId.
         """
         self.watcher.run(self.callback)
+
+
+class Uploader():
+    """Class for handling uploads to the Google cloud storage bucket.
+    """
+    HEARTBEAT_PREFIX = "heartbeats"
+
+    def __init__(self):
+        if not HAS_GOOGLE_STORAGE:
+            from lsst.summit.utils.utils import GOOGLE_CLOUD_MISSING_MSG
+            raise RuntimeError(GOOGLE_CLOUD_MISSING_MSG)
+        self.client = storage.Client()
+        self.bucket = self.client.get_bucket('rubintv_data')
+        self.log = _LOG.getChild("googleUploader")
+
+    def uploadHeartbeat(self, channel, flatlinePeriod):
+        """Upload a heartbeat for the specified channel to the bucket.
+
+        Parameters
+        ----------
+        channel : `str`
+            The channel name.
+        flatlinePeriod : `float`
+            The period after which to consider the channel dead.
+
+        Returns
+        -------
+        success : `bool`
+            Did the upload succeed?
+        """
+        filename = "/".join([self.HEARTBEAT_PREFIX, channel]) + ".json"
```

Comment:
Are these real paths?  Would `os.path.join` be better here and below?

Reply:
I'm actually conflicted. They're not "real" paths in the strict/traditional sense, as they're the names of `blob`s in GCS, but these do basically function just like paths, though I forget the rules on leading slashes. I think I'd prefer not to change this right now, just in case, but am happy to have a big tidy up of things once we have a TTS version of RubinTV running, if that's OK.

Reply:
Probably somewhere in between, tbh 🙃